test/Options/SimplePutOption.js

```diff
@@ -20,45 +20,278 @@ const { ethers } = require("hardhat");
+      genericDecimals
+    );
+    var { value, nonce, sig } = await signedValue(
+      data_vendor,
+      forexRate * 10 ** genericDecimals
+    );
+    await fxPrice
+      .connect(data_vendor)
+      .setVerifiedValue(value, nonce, ethers.utils.arrayify(sig));
+    const actualFXValue = await fxPrice.connect(data_vendor).getVerifiedValue();
+    console.log(
+      "FX price feed created with ticker " +
+        (await fxPrice.getTicker()) +
+        " with initial rate " +
+        actualFXValue / 10 ** genericDecimals
+    );
+
+    equityPrice = await EquityPrice.connect(data_vendor).deploy(
+      "Underlying",
+      "Underlying price",
+      data_vendor.address,
+      genericDecimals
+    );
+    var { value, nonce, sig } = await signedValue(data_vendor, underlyingPrice);
+    await equityPrice
+      .connect(data_vendor)
+      .setVerifiedValue(value, nonce, ethers.utils.arrayify(sig));
+    const actualValue = await equityPrice
+      .connect(data_vendor)
+      .getVerifiedValue();
+    console.log(
+      "Underlying price feed created with ticker " +
+        (await equityPrice.getTicker()) +
+        " with initial price " +
+        actualValue / 10 ** genericDecimals
+    );
+    SimplePutOption = await ethers.getContractFactory("SimplePutOption");
+  });
+
+  async function signedValue(signer, value) {
+    // Create a signed message
+    const nonce = Math.floor(Date.now());
+    const secretMessage = ethers.utils.solidityPack(
+      ["uint256", "uint256"],
+      [value, nonce]
+    );
+    const secretMessageHash = ethers.utils.keccak256(secretMessage);
+    const sig = await signer.signMessage(
+      ethers.utils.arrayify(secretMessageHash)
+    ); // Now signed as secure source
+    return { value, nonce, sig };
+  }
+
+  async function deployPriceContracts() {
+    // Deploy Library
+    const VerifySignerLib = await ethers.getContractFactory("VerifySigner");
+    const verifySignerLib = await VerifySignerLib.deploy();
+    await verifySignerLib.deployed();
+
+    const EquityPrice = await ethers.getContractFactory("EquityPrice", {
+      libraries: {
+        VerifySigner: verifySignerLib.address,
+      },
+    });
+
+    const FXPrice = await ethers.getContractFactory("FXPrice", {
+      libraries: {
+        VerifySigner: verifySignerLib.address,
+      },
+    });
+
+    return { EquityPrice, FXPrice };
+  }
+
+  async function updateEquityPrice(underlyingPrice) {
+    var { value, nonce, sig } = await signedValue(data_vendor, underlyingPrice);
+    await equityPrice
+      .connect(data_vendor)
+      .setVerifiedValue(value, nonce, ethers.utils.arrayify(sig));
+    const actualValue = await equityPrice
+      .connect(data_vendor)
+      .getVerifiedValue();
+    console.log(
+      "Underlying price with ticker " +
+        (await equityPrice.getTicker()) +
+        " updated to " +
+        actualValue / 10 ** genericDecimals
+    );
+  }
+
+  async function updateForexRate(currentForexRate) {
```

Comment:
Function name is updateForexRate - but seems to call the getVerifiedValue() rather than set ?